### PR TITLE
Fix tox mypy installation

### PR DESCRIPTION
**Pull request by LudvigHz**
Mypy needs all deps to check types.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -6,7 +6,10 @@ skipsdist = True
 basepython = python3.11
 allowlist_externals = poetry
 commands_pre =
-    poetry install --only mypy,formatting,flake8
+    isort: poetry install --only formatting
+    flake8: poetry install --only flake8
+    black: poetry install --only formatting
+    mypy: poetry install --with prod --without formatting,flake8
```
Comment (ivarnakken) [inline tox.ini:12]:
```suggestion
    mypy: poetry install --only mypy
```

**Review by ivarnakken — COMMENTED** [review]
`mypy` seems to run fine through `tox` for me? Why do you mean there's a bug?

Are you saying `mypy` will wrongly check files if it does not have access to the deps - makes sense. I suppose it works for me because I already have the deps in the venv. But, you can't run `mypy` through `tox` without `tox` though, so you kind of _have_ to install the default deps first. How have you encountered this?

**Comment by LudvigHz** [comment]
> mypy seems to run fine through tox for me? Why do you mean there's a bug?

What? It's not working at all. It's also failing in all CI runs?

> Are you saying mypy will wrongly check files if it does not have access to the deps - makes sense. I suppose it works for me because I already have the deps in the venv. But, you can't run mypy through tox without tox though, so you kind of have to install the default deps first. How have you encountered this?

Well, mypy needs all deps in the project, because they contain the types? So if tox can't resolve package imports it can't check the types. I don't understand how it could work without installing any dependencies.

You can't have access to the venv, cause tox should create an isolated env with the deps. So even more strange if it works for you :thinking:.

TLDR; As it now it shouldn't work for anyone? It certainly does not work for me or in CI.

**Review by ivarnakken — APPROVED** [review]
python venvs is a mess 🤷🏼
